storageClient/storageClient_test.go

```diff
@@ -2051,6 +2053,94 @@ func TestListGroupMembersReturnsSuccess(t *testing.T) {
 	}
 }
 
+func TestAccountIsLockedByFailingToGetNoteWithIncorrectTozIDEACP(t *testing.T) {
+	// Make request through cyclops to test tozny header is parsed properly
+	registrationClient := accountClient.New(e3dbClients.ClientConfig{Host: cyclopsServiceHost}) // empty account host to make registration request
+	queenClientConfig, _, err := test.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), cyclopsServiceHost)
+	if err != nil {
+		t.Fatalf("Could not register account %s\n", err)
+	}
+	StorageClient := storageClient.New(queenClientConfig)
+	IdentityClient := identityClient.New(queenClientConfig)
+	realmName := fmt.Sprintf("TestAccountIsLockedByFailingToGetNoteWithIncorrectTozIDEACP%d", time.Now().Unix())
+	sovereignName := "Yassqueen"
+	params := identityClient.CreateRealmRequest{
+		RealmName:     realmName,
+		SovereignName: sovereignName,
+	}
+	realm := CreateRealmWithParams(t, IdentityClient, params)
+	defer IdentityClient.DeleteRealm(testCtx, realm.Name)
+
+	noteIDString := uuid.New().String()
+	originalData := map[string]string{"password": "bad_password"}
+	// Write note the first time
+	noteToWrite, err := generateNoteWithData(queenClientConfig.SigningKeys.Public.Material, originalData, noteIDString, queenClientConfig)
+	if err != nil {
+		t.Fatalf("unable to generate note with config %+v\n", queenClientConfig)
+	}
+
+	// Add EACP for the note
+	noteToWrite.EACPS = &storageClient.EACP{
+		TozIDEACP: &storageClient.TozIDEACP{
+			RealmName: realm.Name,
+		},
+	}
+
+	noteWithNoteID, err := StorageClient.WriteNote(testCtx, *noteToWrite)
+	if err != nil {
+		t.Fatalf("unable to write note %+v, to storage servive %s\n", noteToWrite, err)
+	}
+
+	// Attempt to get the same note with a different TozIDEACP
+	eacpParam := make(map[string]string)
+	eacpParam["tozid_eacp"] = "this is the wrong tozid_eacp"
+	for i := 0; i <= identityLoginRetries; i++ {
+		_, err = StorageClient.ReadNote(testCtx, noteWithNoteID.NoteID, nil)
+		if err == nil {
+			t.Fatalf("Expected read note to fail due to incorrect TozIDEACP for note %+v\n", noteToWrite)
+		}
+	}
+
+	// Account should become locked. Attempt to get read the note.
+	_, err = StorageClient.ReadNote(testCtx, noteWithNoteID.NoteID, nil)
```

Comment:
is there a  way to distinguish that its failing due to a lock on the account rather than because it has the wrong EACP? maybe you could lookup the lock itself to make sure it exists?

Reply:
Good call, I think the e3db-client will just pass back the response from the server: "Expectation Failed" or "Unauthorized" in this case

Maybe we should check it explicitly? 
- `if strings.Contains(fmt.Sprint(err), "Expectation Failed")` for lock check and
- `if strings.Contains(fmt.Sprint(err), "Unauthorized")` for 401

Reply:
the string check could work. checking the status code would be more effective. i probably don't have the context to know if you can do this, but @carrala did a status check in identity service [here](https://github.com/tozny/identity-management-service/pull/222/files#diff-d00f75cf1660f5695f7b02430161785c4c36f95d45e706282cf190be5109c154R7322-R7329),

if you change the loop code to check the error, be aware that _i think_ because it is `<=` you may get the locked error on the final iteration of the loop

Reply:
and by "you" i mean @Mrhea haha did not read your username 😂

Reply:
> checking the status code would be more effective.

yeah that makes way more sense, I'm not sure what info is available in the error object

Reply:
@team++ for the catch, @rosasck ++ for the robust assertion